api/src/neo4j/queries/reaction.js

```diff
@@ -2,6 +2,15 @@ import querySingleResult from 'neo4j/queryHandlers/single';
 import reformatExternalDbs from 'neo4j/shared/formatter';
 import parseParams from 'neo4j/shared/helper';
 
+const formatGeneRule = ({ geneRule, genes }) => geneRule.split(/\s+/).map(w => {
+  if (w.match(/and|or/)) {
+    return w;
+  }
+
+  const gene = genes.find(g => g.id === w);
+  return gene.name;
+}).join(' ');
+
```

Comment:
Wouldn't we want to offload this client-side?

Reply:
We could but since our API is usable outside of the frontend don't we want the response to have the formatting directly? Either way is fine, just let me know.

Reply:
I guess in terms of efficiency one could argue both ways.
For the external API users though I think it's best to also provide an untouched version of the GPR, since that may be used in different workflows that expect the GPR to contain gene IDs.

Reply:
Ok sounds good. I will completely revert this in the backend then.

Reply:
Addressed in https://chalmersysbiocsb.atlassian.net/browse/MA-910.